qdrant_client/connection.py

```diff
@@ -130,6 +132,17 @@ def header_adder_interceptor(
     new_metadata: list[tuple[str, str]],
     auth_token_provider: Optional[Callable[[], str]] = None,
 ) -> _GenericClientInterceptor:
+    def process_response(response: Any) -> Any:
+        if response.code() == grpc.StatusCode.RESOURCE_EXHAUSTED:
+            retry_after = next(
+                (item.value for item in response.trailing_metadata() if item.key == "retry-after"),
+                None,
+            )
```

Comment:
it depends on the amount of possible options in `trailing_metadata`, but it might be better to rewrite it with a loop with `break` statement in order to reduce the number of iterations since this function is called on every grpc request